waitfor.go

```diff
@@ -237,7 +240,9 @@ func (client Client) WaitForLKEClusterConditions(
 			case <-ticker.C:
 				result, err := condition(ctx, conditionOptions)
 				if err != nil {
-					return err
+					if !options.Retry {
+						return err
+					}
```

Comment:
We might want to add a log statement here so we can more easily debug issues that may be retried on.

```suggestion
					}

					log.Printf("[WARN] Ignoring WaitForLKEClusterConditions conditional error %s", err)
```